backend/src/test/relay/fetchBlocksInBatches.test.ts

```diff
@@ -0,0 +1,98 @@
+import * as tap from 'tap';
+
+import { TxBlock } from '../../types';
+import { setup } from './common';
+
+tap.test('Relay module - fetchBlocksInBatches method', async (t) => {
+  const {
+    Relay,
+    defaultRelayParams,
+    finalizedBlockNumber,
+    relayWithDefaultParams,
+  } = await setup();
+
+  const nextBlockToProcess = 0;
+  const lastFinalizedBlockNumber = () => finalizedBlockNumber;
+
+  tap.test('should yield block number next to the block number of the block in the batch', async (t) => {
+    const batchesGenerator = relayWithDefaultParams['fetchBlocksInBatches'](
```

Comment:
It is a bit annoying to test APIs like this. Do you actually need to test private implementation details?

Reply:
I agree, it looks ugly and normally I would test only public methods, but these private methods are pretty complex, so I think having tests for them is beneficial